src/asset/audio.rs

```diff
@@ -52,18 +75,44 @@ impl Audio {
     }
 
     pub fn load_data(&mut self, bytes: Vec<u8>) {
+        let samples = match self.kind {
+            AudioFileKind::Ogg => self.samples_ogg(bytes),
+            AudioFileKind::Wav => self.samples_wav(bytes),
+        };
+        *self.samples.write().unwrap() = samples;
+    }
+
+    pub fn samples_wav(&mut self, bytes: Vec<u8>) -> Vec<f32> {
         let (header, data) = wav::read(&mut std::io::Cursor::new(bytes)).unwrap();
-        if header.sampling_rate as i32 != crate::audio::SAMPLE_RATE {
+        if header.sampling_rate != crate::audio::SAMPLE_RATE as u32 {
             println!(
                 "warn: {} contains non-supported sample rate {}\nwarn: only 48000 is currently supported",
                  self.data.file.display(),
                  header.sampling_rate,
             );
         }
-        let mut samples = self.samples.write().unwrap();
         match data {
-            wav::BitDepth::ThirtyTwoFloat(data) => *samples = data,
+            wav::BitDepth::ThirtyTwoFloat(data) => data,
             _ => todo!("Only WAV containing floats are currently supported"),
         }
     }
+
+    pub fn samples_ogg(&mut self, bytes: Vec<u8>) -> Vec<f32> {
```

Comment:
Isn't the thing with OGG that we can stream in the audio we need. And don't have everything sorted in ram at all times?

Is that something we want?

Reply:
Yes, that sounds nice. But, small PRs! #64